### ISSUE TITLE
spacy ner scorer return all zeros and duplicate labels

### ISSUE THREAD

**Issue by ivan90dg** [explosion/spaCy#5318]
Hi all,

I am trying to get scores on my test set using Scorer. Here is my simple code:

```
test_set = [
    ('Your 40577 is finished', [(6,11,'LABEL1')] ),
    ('Finished with SODE20915', [(14,23,'LABEL2')] )
]
```
```
scorer = Scorer()
for text,annotin test_set:
    doc_gold_text = nlp(text)  # here i tried also: doc_gold_text = nlp.make_doc(txt)
    gold = GoldParse(doc_gold_text, entities=annotin)
    pred_value = nlp(txt)
   #print(gold.words, gold.tags, gold.labels, gold.ner)
    scorer.score(pred_value, gold)
print(scorer.scores)
```

And what I am getting as output is all zeros and duplicate LABEL1 as 'LABEL1' and "'LABEL1'":

`{'uas': 0.0, 'las': 0.0, 'las_per_type': {'': {'p': 0.0, 'r': 0.0, 'f': 0.0}}, 'ents_p': 0.0, 'ents_r': 0.0, 'ents_f': 0.0, 'ents_per_type': {"'LABEL1'": {'p': 0.0, 'r': 0.0, 'f': 0.0}, 'LABEL1': {'p': 0.0, 'r': 0.0, 'f': 0.0},"'LABEL2'": {'p': 0.0, 'r': 0.0, 'f': 0.0}, 'LABEL2': {'p': 0.0, 'r': 0.0, 'f': 0.0}}, 'tags_acc': 0.0, 'token_acc': 100.0, 'textcat_score': 0.0, 'textcats_per_cat': {}}`

I don't understand why I have duplicate labels? And model made correct predictions in both sentences, but I have P,R,F1 scores equal to 0? Maybe i missed something?

When I print GoldParse object (uncomment print line in the loop) I am getting this:

```
['Your', '40577','is','finished'] [None, None,None,None] [None, None,None,None] ['O', 'U-CNTNUM','O','O']
['Finished ', 'with', '41735'] [None, None, None] [None, None, None] ['O', 'O', 'U-CNTNUM']

```

I checked my train data it seems ok. Here is example of train data:

```
[('This paper BIME 13935 will be done soon', {'entities': [(11, 21, 'LABEL2')]}), 
('What is status of 43391', {'entities': [(18, 23, 'LABEL1')]})]
```

Model is trained on GPU using minibatches.

## Your Environment
* Operating System: Linux Ubuntu
* Python Version Used: python 3.7
* spaCy Version Used: spacy 2.2.4
* Environment Information: GPU

**Comment by adrianeboyd** [comment]
I'm not sure where the labels with quotes would have come from except from your training data. Can you double-check the formatting of all the labels in your training data to see if that's where the extra quotes are coming from?

You can also try running `spacy.gold.biluo_tags_from_offsets(nlp.make_doc(text), entities)` for each text in your training data to see if there are any unexpected results or formatting, since this is the same function that's used to convert the data internally.

**Comment by svlandeg** [comment]
@adrianeboyd : these zeros look like #5322 ?

**Comment by adrianeboyd** [comment]
@svlandeg: Without all the details it's hard to know for sure, but I think it's a separate issue.

**Comment by ivan90dg** [comment]
I found what caused duplicate label names, i fixed it in my code. But still have problem with zeros for P,R,F1. Here is complete simple example which I used for training and testing:

```
import thinc_gpu_ops
thinc_gpu_ops.AVAILABLE

import spacy 
spacy.prefer_gpu()
from spacy.util import minibatch, compounding
from spacy.gold import GoldParse
from spacy.scorer import Scorer

TRAIN_DATA = [
    ("Who is Shaka Khan?", {"entities": [(7, 17, "PERSON")]}),
    ("My name is Ivan", {"entities": [(11, 15, "PERSON")]}),
    ("I like London and Berlin.", {"entities": [(7, 13, "LOC"), (18, 24, "LOC")]}),
    ("Sara went to the supermarket", {"entities": [(0, 4, "PERSON")]}),
    ("When we were on vacation we went to Verona to see some interesting things there", {"entities": [(36, 42, "LOC")]}),
]

def train_spacy(nlp, data,iterations):
    TRAIN_DATA = data
    if 'ner' not in nlp.pipe_names:
        ner = nlp.create_pipe('ner')
        nlp.add_pipe(ner, last=True)
    else:
        ner = nlp.get_pipe('ner')

    # add labels
    for _, annotations in TRAIN_DATA:
         for ent in annotations.get('entities'):
            ner.add_label(ent[2])

    # get names of other pipes to disable them during training
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
    with nlp.disable_pipes(*other_pipes):  # only train NER
        optimizer = nlp.begin_training()
        for itn in range(iterations):
            print("Statring iteration " + str(itn))
            random.shuffle(TRAIN_DATA)
            losses = {}
            batches = minibatch(TRAIN_DATA, size=compounding(4.0,32.0,1.001))
            for batch in batches:
                text, annotations = zip(*batch)
                nlp.update(
                    text,  # batch of texts
                    annotations,  # batch of annotations
                    drop=0.2,  # dropout - make it harder to memorise data
                    sgd=optimizer,  # callable to update weights
                    losses=losses)
            print(losses)
    return nlp


nlp = spacy.load("en_core_web_sm")
model = train_spacy(nlp, TRAIN_DATA, 20)

test_examples = [
    ("Who was Hamlet?", [(8,14,'PERSON')]),
    ("Have you ever met Rome?", [(18,22,'LOC')])
]

scorer = Scorer()
for text,annot in test_examples:
    doc_gold_text = model.make_doc(text)
    gold = GoldParse(doc_gold_text, entities=annot)
    pred_value = model(text)
    scorer.score(doc_gold_text, gold)
    
print(scorer.scores)
```

And I am getting this:
```
{'uas': 0.0, 'las': 0.0, 'las_per_type': {'': {'p': 0.0, 'r': 0.0, 'f': 0.0}}, 'ents_p': 0.0, 'ents_r': 0.0, 'ents_f': 0.0, 'ents_per_type': {'PERSON': {'p': 0.0, 'r': 0.0, 'f': 0.0}, 'LOC': {'p': 0.0, 'r': 0.0, 'f': 0.0}}, 'tags_acc': 0.0, 'token_acc': 100.0, 'textcat_score': 0.0, 'textcats_per_cat': {}}
```

When I am try what you suggest:
```
for text,annot in test_examples:
    b = spacy.gold.biluo_tags_from_offsets(nlp.make_doc(text), annot)
    print(b)
```
It looks fine, this is result:
```
['O', 'O', 'U-PERSON', 'O']
['O', 'O', 'O', 'O', 'U-LOC', 'O']
```

I am not sure what I am doing wrong here to get all metrics equal to 0.

**Comment by adrianeboyd** [comment]
Thanks, a full example makes this much easier to troubleshoot!

This may not be the same problem as in your original code where you don't have the same mistake, but in the example above the mistake is here:

```python
scorer.score(doc_gold_text, gold)
```

You're giving the scorer the blank tokens-only document instead of the annotated one. You need to have:

```python
scorer.score(pred_value, gold)
```

With this change you can see non-zero scores for the demo example:

```
'ents_p': 50.0, 'ents_r': 50.0, 'ents_f': 50.0, 'ents_per_type': {'LOC': {'p': 50.0, 'r': 100.0, 'f': 66.66666666666666}, 'PERSON': {'p': 0.0, 'r': 0.0, 'f': 0.0}}
```

**Comment by ivan90dg** [comment]
Oh I missed that, now it works. Great, thank you!
